src/analyses/mutexAnalysis.ml

```diff
@@ -9,11 +9,13 @@ open Prelude.Ana
 open Analyses
 
 
+(* TODO: remove SpecParam because only one implementation? *)
 module type SpecParam =
 sig
   module G: Lattice.S
   val effect_fun: ?write:bool -> Lockset.t -> G.t
   val check_fun: ?write:bool -> Lockset.t -> G.t
+  val export: ?write:bool -> G.t -> Mutexes.t
 end
```

Comment:
Mutex analysis is a functor parametrized about such global domain, of which there is only a single implementation. Maybe it'd be clearer to just inline this instead of adding various conversion functions for everything?
It was functorized because OSEK used a different implementation, but that's now gone.

Reply:
I agree, we should do so!

Reply:
Done!

There's also the opportunity to refactor the global invariant from (readwrite, write) to (read, write), which is simply more general, or even a map on all `AccessKind`s, but that can wait since it's not really necessary or useful right now.